src/services/users/update-user.js

```diff
@@ -0,0 +1,64 @@
+const db = require("../../db")
+const customError = require("../../lib/custom-error")
+const { UserNotFoundError } = require("../errors")
+
+/**
+ * 
+ * @param {Number} id - id of the user to be updated
+ * @param {Object} cols - columns to be updated
+ * @returns {String} - query to be executed
+ */
+const constructQuery = (id, cols) => {
+
+	// Setup static beginning of query
+	const query = ['UPDATE users']
+	query.push('SET')
+  
+	// Create another array storing each set command
+	// and assigning a number value for parameterized query
+	const set = []
+	Object.keys(cols).forEach((key, i) => {
+	  set.push(`"${key}" = ($${(i + 1)})`) 
+	})
+	query.push(set.join(', '))
+  
+	// Add the WHERE statement to look up by id
+	query.push(`WHERE id = ${id} RETURNING  "id",
+	"firstName", "lastName", "email",
+	"gender", "jobRole", "department", 
+	"address", "refreshToken", "profilePictureUrl"` )
+  
+	// Return a complete query string
+	return query.join(' ')
+}
+
+/**
+ * 
+ * @param {Number} id - id of the user to be updated
+ * @param {*} requestBody - reqquest body
```

Comment:
Not getting the `{*}` part yet and also a small typo

```suggestion
 * @param {*} requestBody - request body
```

Reply:
It was a mistake, will update it